api/oci/extensions/repositories/artifactset/artifactset.go

```diff
@@ -204,26 +206,123 @@ func (a *namespaceContainer) AddTags(digest digest.Digest, tags ...string) error
 	defer a.base.Unlock()
 
 	idx := a.GetIndex()
-	for i, e := range idx.Manifests {
-		if e.Digest == digest {
-			if e.Annotations == nil {
-				e.Annotations = map[string]string{}
-				idx.Manifests[i].Annotations = e.Annotations
-			}
-			cur := RetrieveTags(e.Annotations)
-			if cur != "" {
-				cur = strings.Join(append([]string{cur}, tags...), ",")
-			} else {
-				cur = strings.Join(tags, ",")
-			}
-			e.Annotations[TAGS_ANNOTATION] = cur
-			if a.base.FileSystemBlobAccess.Access().GetInfo().GetDescriptorFileName() == OCIArtifactSetDescriptorFileName {
-				e.Annotations[OCITAG_ANNOTATION] = tags[0]
+	isOCI := a.base.FileSystemBlobAccess.
+		Access().
+		GetInfo().
+		GetDescriptorFileName() == OCIArtifactSetDescriptorFileName
+
+	// Collect all descriptors for this digest (there may already be multiple).
+	var descIdxs []int
+	for i := range idx.Manifests {
+		if idx.Manifests[i].Digest == digest {
+			descIdxs = append(descIdxs, i)
+		}
+	}
+	if len(descIdxs) == 0 {
+		return errors.ErrUnknown(cpi.KIND_OCIARTIFACT, digest.String())
+	}
+
+	// Build complete tag set from:
+	//   - existing TAGS_ANNOTATION on any descriptor for this digest
+	//   - incoming tags
+	tagSet := map[string]struct{}{}
```

Comment:
From what I can see this could also be just a `[]string`. Am I missing something?

Reply:
the point of this was that i am able to unify both tags from the vararg and the annotation without having to deduplicate